cpp/include/cudf/detail/copy_if.cuh

```diff
@@ -116,11 +116,14 @@ __global__ void scatter_kernel(cudf::mutable_column_device_view output_view,
   for (int i = 0; i < per_thread; i++) {
     bool mask_true = (tid < size) && filter(tid);
 
-    block_sum = 0;
+
+    cudf::size_type tmp_block_sum = 0;
     // get output location using a scan of the mask result
     const cudf::size_type local_index = block_scan_mask<block_size>(mask_true,
-                                                                   block_sum);
-
+                                                                    tmp_block_sum);
```

Comment:
Can you explain this change? Every thread's block_sum is the same, so checking for thread zero is just an unnecessary branch and unnecessary complexity. I don't see a need for tmp_block_sum.

Reply:
The issue is when the per_thread is > 1, the block_sum in next iteration returns 0.

Reply:
I see. Yes, there is a bug here. However I think you should get rid of the `if (threadIdx.x == 0)` -- it's unnecessary divergence and each thread has its own identical copy of this variable anyway, so it's likely cheaper to just do the add in every thread than to do the branch in every thread.